From a9fd170a9ab1278585a4e7415cef5ca3e84e73eb Mon Sep 17 00:00:00 2001
From: Karthik Kambatla <kasha@apache.org>
Date: Mon, 24 Oct 2016 22:19:14 -0700
Subject: [PATCH 1964/2748] YARN-5754. Null check missing for earliest in
 FifoPolicy. (Yufei Gu via kasha)

(cherry picked from commit a71fc81655cd5382d354674dd06570ba49753688)
(cherry picked from commit ab9105d81449dd466022e799fab6b27d6e9ff01e)

Change-Id: I3bdff76c4674f61317e1926028ac4c5ccb5ce4df
---
 .../scheduler/fair/policies/FifoPolicy.java        |    5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-resourcemanager/src/main/java/org/apache/hadoop/yarn/server/resourcemanager/scheduler/fair/policies/FifoPolicy.java b/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-resourcemanager/src/main/java/org/apache/hadoop/yarn/server/resourcemanager/scheduler/fair/policies/FifoPolicy.java
index 49abde0..3bea983 100644
--- a/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-resourcemanager/src/main/java/org/apache/hadoop/yarn/server/resourcemanager/scheduler/fair/policies/FifoPolicy.java
+++ b/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-resourcemanager/src/main/java/org/apache/hadoop/yarn/server/resourcemanager/scheduler/fair/policies/FifoPolicy.java
@@ -96,7 +96,10 @@ public void computeShares(Collection<? extends Schedulable> schedulables,
         earliest = schedulable;
       }
     }
-    earliest.setFairShare(Resources.clone(totalResources));
+
+    if (earliest != null) {
+      earliest.setFairShare(Resources.clone(totalResources));
+    }
   }
 
   @Override
-- 
1.7.9.5

